consensus/service/src/timestamp_validator.rs

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) 2023 The MobileCoin Foundation
+
+use displaydoc::Display;
+
+/// Provides logic for validating and ensuring that a timestamp exists when
+/// combining `ConsensusValues`.
+
+const MAX_TIMESTAMP_AGE: u64 = 30 * 1000; // 30 seconds
+
+/// Errors related to validating timestamps
+#[derive(Debug, Display, Clone, PartialEq)]
+pub enum Error {
+    /// The timestamp is in the future now{0}, timestamp{1}
+    InFuture(u64, u64),
+    /** The timestamp is older than the last bock: last_bock timestamp{0},
+    timestamp{1} */
+    OlderThanLastBlock(u64, u64),
+    /// The timestamp is too far in the past now{0}, timestamp{1}
```

Comment:
Similarly, combining the timestamp with the word before it seems a bit weird to me

Reply:
conversation brought over to, https://github.com/mobilecoinfoundation/mobilecoin/pull/3733/files#r1399806093